### PR TITLE
Html support

### PR DESCRIPTION
**Pull request by Hiswe**
This will allow to beautify HTML as well

### CODE REVIEW

**Comment by contra** [comment]
On one hand I like that it can handle multiple extensions, it certainly makes things simpler (one plugin to beautify all files). On the other hand though, it does go against our original "plugins do one thing" principle. In general we've started to be more permissive on those principles where it makes sense, so this could be one of those cases.

**Comment by Hiswe** [comment]
I agree, it's just a wrapper around `js-beautify`, so it still does one thing.

In my opinion it feels confusing that `gulp-beautify` doesn't beautify all the files supported by `js-beautify` (I was confused and need to read the source code to understand why my html files were ignored)
And also it makes users sure that they will use the same `js-beautify` version whenever they call `gulp-beautify`.

**Comment by phated** [comment]
The way I've been thinking about plugins is that they shouldn't be guarded upon extension name (for example, what if your consumers are using the `.mjs` or `.jsx for their files?) and those consumers should filter by file extension previous to said plugin.  The general use case is that people are only globbing valid files anyway.

**Comment by Hiswe** [comment]
You're right.

Maybe it can  support a `type` option?

```js
gulp.task("beautify", function() {
  gulp
    .src("./src/*.mjs")
    .pipe(beautify({ 
      type: "html", 
      indent_size: 2,
    }))
    .pipe(gulp.dest("./public/"));
});
```

or [language overrides](https://github.com/beautify-web/js-beautify#setting-inheritance-and-language-specific-overrides) are serving that purpose if we don't restrict `js-beautify` to the js parser?

**Comment by phated** [comment]
Nah, I think it should just not have a guard at all. If js-beautify can handle other file formats than JS, everything should be passed through verbatim and options should just be handed off to the library.

**Comment by phated** [comment]
We should also dump rcLoader while we're in here.

**Comment by Hiswe** [comment]
I'll try but I'm not sure `js-beautify` can use the right beautyfier by himself

removing rcLoader shouldn't be a problem

**Comment by Hiswe** [comment]
@phated so there is no way to let *js-beautify* [decide automatically which beautifier](https://github.com/beautify-web/js-beautify#css--html) it should use.

Since we don't want to rely on an extension mapping, we could rely on a `beautifier`  parameter (either `js`(default), `css`, or `html`)

Or leave the plugin as it is and handle only JS.
other plugin [(like gulp-html-beautify)](https://www.npmjs.com/package/gulp-html-beautify) could handle the other beautifiers…

**Comment by phated** [comment]
I'd rather have 4 exports:
* default = js
* js
* html
* css

Each could take just the options they support. Thoughts?

**Comment by Hiswe** [comment]
@phated it seems the best idea as it mirrors js-beautify exports

**Comment by Hiswe** [comment]
@phated should be ok

File: README.md [hunk 1]
```diff
@@ -1,30 +1,50 @@
...
+As with js-beautify you can use it for [HTML & CSS](https://github.com/beautify-web/js-beautify#css--html):
+
+```javascript
+var beautifyHtml = require('gulp-beautify').html;
+var beautifyCss = require('gulp-beautify').css;
+
+gulp.task('beautify-html', function() {
+  gulp
+    .src('./src/*.html')
+    .pipe(beautifyHtml({ indent_size: 2 }))
```
Comment (phated) [inline README.md]: Please make these examples `beautify.html(...)` and `beautify.css(...)`

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 sudo: false
 language: node_js
 node_js:
-  - "0.10"
+  - '10'
+  - '8'
+  - '6'
```
Comment (phated) [inline .travis.yml]: please add 0.10, 0.12 and 4 to the build matrix

File: README.md [hunk 3]
```diff
@@ -1,30 +1,50 @@
...
 
 ```javascript
 var beautify = require('gulp-beautify');
 
 gulp.task('beautify', function() {
-  gulp.src('./src/*.js')
-    .pipe(beautify({indent_size: 2}))
-    .pipe(gulp.dest('./public/'))
+  gulp
+    .src('./src/*.js')
```
Comment (phated) [inline README.md]: Should we add the `beautify.js(...)` example and then move this below all the other ones and mark it as the old pattern?

Comment (Hiswe) [reply README.md]: That's a good idea. I will be more inline with [js-beautify documentation](https://github.com/beautify-web/js-beautify#nodejs-javascript-1)

File: .travis.yml (hunk 2 above)
Comment (Hiswe) [reply .travis.yml]: @phated I've been able to run the tests on node 4.
But it fails on Node 0.12.18:

```
node_modules/mocha/bin/mocha:10
const spawn = require('child_process').spawn;
^^^^^
SyntaxError: Use of const in strict mode.
```
Is it possible to drop 0.10 & 0.12?

Comment (phated) [reply .travis.yml]: reduce the mocha version to `^3` and it works. Mocha isn't a critical dep.

Comment (Hiswe) [reply .travis.yml]: @phated sorry for the delay.
Ok done & working.

**Comment by phated** [comment]
@Hiswe just one more issue.  Sorry I didn't catch it before.

**Comment by phated** [comment]
@contra can you add me to this package on npm so I can publish once that last thing is done?

File: package.json [hunk 4]
```diff
@@ -11,13 +11,13 @@
     "beautify"
   ],
   "dependencies": {
-    "js-beautify": "^1.5.10",
-    "rcloader": "^0.2.1",
-    "through2": "^2.0.0"
+    "js-beautify": "^1.8.8",
+    "through2": "^3.0.0"
```
[the remark this answers is not in the document]
Comment (Hiswe) [reply package.json]: @phated done

[the remark this answers is not in the document]
Comment (phated) [reply package.json]: Thanks, just waiting on publish permissions and then I'll get this released.

**Comment by phated** [comment]
@Hiswe thanks for all the work on this!  I've published your changes as 3.0.0 - sorry for the delay!

**Comment by Hiswe** [comment]
@phated Thanks for all the insights!

**Pull request merged**